src/components/Docs/components/ForeignCoinsTable.tsx

```diff
@@ -34,13 +48,76 @@ const formatString = (str: string) => {
     .join(" ");
 };
 
+const addTokenToMetaMask = async (token: ForeignCoin & { chainName: string }) => {
+  if (!window.ethereum) {
+    alert("Please install an EVM wallet like MetaMask to add tokens to your wallet");
+    return;
+  }
+
+  try {
+    const wasAdded = await window.ethereum.request({
+      method: "wallet_watchAsset",
+      params: {
+        type: "ERC20",
+        options: {
+          address: token.zrc20_contract_address,
+          symbol: token.symbol,
+          decimals: token.decimals,
+        },
+      },
+    });
+
+    if (wasAdded) {
+      console.log("Token was added to MetaMask");
+    }
+  } catch (error) {
+    console.error("Error adding token to MetaMask:", error);
+  }
+};
+
 export const ForeignCoinsTable = () => {
   const [mainnetCoins, setMainnetCoins] = useState<(ForeignCoin & { chainName: string })[]>([]);
   const [testnetCoins, setTestnetCoins] = useState<(ForeignCoin & { chainName: string })[]>([]);
-
+  const [currentChainId, setCurrentChainId] = useState<string | null>(null);
   const [isLoading, setIsLoading] = useState(true);
   const [activeTab, setActiveTab] = useState(networkTypeTabs[0]);
 
+  // Check if the current network matches the active tab
+  const isCorrectNetwork = useMemo(() => {
+    if (!currentChainId) return false;
+    if (activeTab.networkType === "mainnet") {
+      return currentChainId === ZETACHAIN_MAINNET_ID;
+    }
+    return currentChainId === ZETACHAIN_TESTNET_ID;
+  }, [currentChainId, activeTab.networkType]);
+
+  // Listen for chain changes
+  useEffect(() => {
```

Comment:
I believe this can probably live in a custom hook, it sounds like it could be beneficial to have this available in other components.